code/game/area/Space Station 13 areas.dm

```diff
@@ -41,6 +41,8 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 	var/used_environ = 0
 
 	var/has_gravity = 1
+	var/cached_gravity = 1		//stores updated has_gravity even if it's blocked
```

Comment:
Is there really a need for this cached gravity var? 

Couldn't the gravity generator just tell the area to update its gravity to what it should be (by checking for the presence of a gravity generator on the ship) when it moves?

Reply:
It needed for the case when gravity generator changes area's gravity while it's blocked.
This is much better than looking for the gravity generator on map.

Reply:
Its an example of a special case (antigrav generators) using a generic and common object (area datums) to store data

Searching for a gravity generator on the map may be a little less efficient, but it'd be a better design.
In any case, for all kinds of practical purposes, we definitely need to have a reset_gravity proc. Where an area can abandon whatever gravity states have been set upon it, re-examine all the factors in the game that would affect it, and set its gravity to what it should be

Reply:
Do you think we need more advanced structure for storing gravity state?

Reply:
I think searching for the factors of anything like that isn't a good thing. It's complex, not very effective and a huge possibility for different bugs.

Reply:
this is not true at all, its both the simplest and least error prone solution